docs/CLI.md

```diff
@@ -0,0 +1,87 @@
+
+You would at least need to add the following lines to your applications in previous releases to get the agent attached and running.  
+
+```python
+from skywalking import agent
+agent.start()
+```
+
+
+Now the SkyWalking Python agent implements a command-line interface that can be utilized to attach the agent to your
+awesome applications during deployment **without changing any application code**, 
+just like the [SkyWalking Java Agent](https://github.com/apache/skywalking).
+
+## Usage
+
+Upon successful installation of the SkyWalking Python agent via pip, 
+a command-line script `sw-python` is installed in your environment(virtual env preferred).
+
+### The run option
+
+Currently, the `sw-python` CLI provides a `run` option, which you can use to execute your applications
+(either begins with the `python` command or Python-based programs like `gunicorn` on your path) 
+just like you invoke them normally, plus a prefix, the following example demonstrates the usage.
+
+If your previous command to run your gunicorn application is:
+
+`gunicorn app.wsgi`
+
+Please change it to:
+
+`sw-python run gunicorn app.wsgi`
+
+The SkyWalking Python agent will startup along with your application shortly.
+
+Note that the command does work with multiprocessing and subprocess as long as the `PYTHONPATH` is inherited, 
+please configure the environment variables configuration accordingly based on the general documentation.
+
+When executing commands with `sw-python run command`, your command's Python interpreter will pick up the SkyWalking loader module.
+
+It is not safe to attach SkyWalking Agent to those commands that resides in another Python installation 
+because incompatible Python versions and mismatched SkyWalking versions can cause problems. 
+Therefore, any attempt to pass a command that uses a different Python interpreter/ environment will not bring up 
+SkyWalking Python Agent even if another SkyWalking Python agent is installed there(no matter the version), 
+and will force exit with an error message indicating the reasoning.
+
+#### Disabling child processes from starting new agents
+
+Sometimes you don't actually need the agent to monitor anything in a child process.
+
+If you do not need the agent to get loaded for application child processes, you can turn off the behavior by setting an environment variable.
+
+`SW_PYTHON_BOOTSTRAP_PROPAGATE` to `False`
+
+Note the auto bootstrap depends on the environment inherited by child processes, 
+thus prepending a new sitecustomize path to or removing the loader path from the `PYTHONPATH` could prevent the agent from loading in a child process. 
+
+### Configuring the agent 
+
+You would normally want to provide additional configurations other than the default ones.
+
+#### Through environment variables
+
+The currently supported method is to provide the environment variables listed 
+in [EnvVars Doc](EnvVars.md) as instructed in the [README](../README.md).
+
+#### Through a sw-config.yaml
```

Comment:
One thing we should note when implementing this is that we've been trying to minimize our dependencies as an agent, so we'd better choose a reasonable file format that Python built-in libs can parse, instead of bringing another library, to avoid version conflicts with user applications, `yaml` is a common case where users may already use and we should not require dependency upgrade or downgrade.